File: CS/SilverlightReports/MainPageViewModel.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.ComponentModel;
using System.Windows.Input;
using DevExpress.Xpf.Core.Commands;
using DevExpress.Xpf.Printing;

namespace SilverlightReports
{
    public class MainPageViewModel : INotifyPropertyChanged
    {
        readonly ReportPreviewModel _previewModel;
        readonly DelegateCommand<object> _categorySubmitCommand;
        readonly ReadOnlyCollection<string> _categories;
        string _category;

        public IDocumentPreviewModel PreviewModel { get { return _previewModel; } }
        public ICommand CategorySubmitCommand { get { return _categorySubmitCommand; } }

        public string Category
        {
            get { return _category; }
            set
            {
                if (value == _category)
                    return;

                _category = value;
                OnPropertyChanged("Category");
            }
        }

        public IEnumerable<string> Categories
        {
            get
            {
                return _categories;
            }
        }

        public MainPageViewModel()
        {
            _previewModel = new ReportPreviewModel() { ServiceUri = "../NorthwindReportService.svc", ReportTypeName = "Reports.ProductReport" };
            _previewModel.RequestDefaultParameterValues += MainPageViewModel_RequestDefaultParameterValues;
            _categorySubmitCommand = new DelegateCommand<object>(CategorySubmitted);
            _categories = new ReadOnlyCollection<string>(new string[] { "Beverages", "Condiments", "Seafood" });
        }

        private void CategorySubmitted(object o)
        {
            if (_previewModel.Parameters.Count > 0)
                _previewModel.Parameters["Category"].Value = Category;

            _previewModel.CreateDocument();
        }

        private void MainPageViewModel_RequestDefaultParameterValues(object sender, EventArgs e)
        {
            _previewModel.Parameters["Category"].Value = Category;
        }

        public event PropertyChangedEventHandler PropertyChanged;
        internal void OnPropertyChanged(string property)
        {
            if (PropertyChanged != null)
                PropertyChanged(this, new PropertyChangedEventArgs(property));
        }
    }
}

File: VB/SilverlightReports/MainPageViewModel.vb
Imports Microsoft.VisualBasic
Imports System.Collections.Generic
Imports System.Collections.ObjectModel
Imports System.ComponentModel
Imports System.Windows.Input
Imports DevExpress.Xpf.Core.Commands
Imports DevExpress.Xpf.Printing
' ...

Namespace SilverlightReports
	Public Class MainPageViewModel
		Implements INotifyPropertyChanged
		Private ReadOnly _previewModel As ReportPreviewModel
		Private ReadOnly _categorySubmitCommand As DelegateCommand(Of Object)
		Private ReadOnly _categories As ReadOnlyCollection(Of String)
		Private _category As String

		Public ReadOnly Property PreviewModel() As IDocumentPreviewModel
			Get
				Return _previewModel
			End Get
		End Property
		Public ReadOnly Property CategorySubmitCommand() As ICommand
			Get
				Return _categorySubmitCommand
			End Get
		End Property

		Public Property Category() As String
			Get
				Return _category
			End Get
			Set(ByVal value As String)
				If value = _category Then
					Return
				End If

				_category = value
				OnPropertyChanged("Category")
			End Set
		End Property

		Public ReadOnly Property Categories() As IEnumerable(Of String)
			Get
				Return _categories
			End Get
		End Property

		Public Sub New()
			_previewModel = New ReportPreviewModel() With {.ServiceUri = "../NorthwindReportService.svc", .ReportName = "Reports.ProductReport"}
			_categorySubmitCommand = New DelegateCommand(Of Object)(AddressOf CategorySubmitted)
			_categories = New ReadOnlyCollection(Of String)(New String() { "Beverages", "Condiments", "Seafood" })
		End Sub

		Private Sub CategorySubmitted(ByVal o As Object)
			_previewModel.Parameters("Category").Value = Category
			_previewModel.CreateDocument()
		End Sub

		Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged
		Friend Sub OnPropertyChanged(ByVal [property] As String)
			RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs([property]))
		End Sub
	End Class
End Namespace
